serde/src/private/de.rs

```diff
@@ -191,17 +191,20 @@ where
 pub mod size_hint {
     use lib::*;
 
+    #[inline]
     pub fn from_bounds<I>(iter: &I) -> Option<usize>
     where
         I: Iterator,
     {
         helper(iter.size_hint())
     }
 
+    #[inline]
     pub fn cautious(hint: Option<usize>) -> usize {
         cmp::min(hint.unwrap_or(0), 4096)
     }
 
+    #[inline]
     fn helper(bounds: (usize, Option<usize>)) -> Option<usize> {
```

Comment:
I believe the point of this helper (as opposed to writing the same code in from_bounds) is to keep the logic from being instantiated for all the different `I` types used with from_bounds. Does inline affect that and is it offset by the benefit of inline here?

Reply:
I haven't measured anything for this, like I said I was just in the neighborhood.  I'd be happy to drop it if you're concerned about it.

Reply:
I don't have an intuition for whether inline would be better or worse here. Let's drop this one and stick to the one you measured a benefit from. Thanks!